### ISSUE TITLE
Send Plans to crypten parties

### ISSUE THREAD

**Issue by LaRiffle** [OpenMined/PySyft#2987]
We'd like to send a built and serialized plan to a crypten party, which will then ask his syft worker to execute it for him, but in crypten mode and with crypten tensors

**Comment by youben11** [comment]
Tried to list some challenges for this. What I have been facing so far:
1. Plans support hooked torch operations exclusively, crypten tensor types aren't supported by syft for the moment.
2. When running a multiparty computation, crypten operations supports the use of the `src` parameter which specify a special behavior at party with rank `src`. If we initialized crypten with a single party group then the build will throw an error `invalid tensor source` as the `src` might be for a party with rank greater than the number of parties initialized.
3. As plans are built by running the function and recording the executed ops, running a single crypten party when more are required will block as synchronization is required between parties to complete the computation.

**Comment by LaRiffle** [comment]
So I think that actually plans would be built using pure torch.tensors, which is ok to record operations in the `run_multiprocessiong`. The only pb is regarding the crypten specific bits:
```

@mpc.run_multiprocess(world_size=2)
def examine_binary_shares():
    x_enc = crypten.load('models/tutorial4_alice_model.pth', dummy_model=dummy_model, src=ALICE)
    
    y = x_enc * 2  # This is easy
    return y
```

Regarding the `crypten.load('path', src...)`, I'd favour replacing it by a simple `syft.search('#tag',src=alice)`

If we do this, we can (at the expense of slightly changing the suntax as shown above), record all the meaning full operations by running the plan only once on the client using cleartext pure torch tensors, and then forward this to the remote workers which hold the crypten workers. (Here syft.search would be recorded by the tracer decorator, but would output probably just a random tensors without really doing a search (we don't want to bother alice))

**Comment by youben11** [comment]
We may require a new plan builder for crypten. I think we can solve issues 2 and 3 listed above by setupping local parties using virtual workers, running the crypten computation and recording operations.

**Comment by LaRiffle** [comment]
Solved by the amazing @gmuraru
